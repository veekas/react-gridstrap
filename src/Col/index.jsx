import React from 'react';
import PropTypes from 'prop-types';
import styled from 'react-emotion';

const Col = ({ children, className, xs, sm, md, lg, xl, ...props }) => {
  const ColDiv = styled('div')`
    box-sizing: border-box;
    position: relative;
  `;

  return (
    <ColDiv className={className} {...props}>
      {children}
    </ColDiv>
  );
}

Col.propTypes = {
  children: PropTypes.node,
  className: PropTypes.string,
  xs: PropTypes.bool,
  sm: PropTypes.bool,
  md: PropTypes.bool,
  lg: PropTypes.bool,
  xl: PropTypes.bool,
};

Col.defaultProps = {
  children: null,
  className: null,
  xs: false,
  sm: false,
  md: false,
  lg: false,
  xl: false,
};

export default Col;
