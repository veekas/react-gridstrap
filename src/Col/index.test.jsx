import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Col from '.';

describe('Col', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Col />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('matches snapshot', () => {
    const tree = renderer
      .create(<Col />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
