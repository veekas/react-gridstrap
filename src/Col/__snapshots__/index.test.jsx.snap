// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Col matches snapshot when passed zero props 1`] = `
<div
  className="css-dtn8f5"
/>
`;
